cloudant-sync-datastore-core/src/test/java/com/cloudant/sync/query/IndexCreatorTest.java

```diff
@@ -162,123 +171,108 @@ public void createIndexWhenIndexNameExistsIdxDefinitionSame() {
-        settings.put("tokenize", "porter");
+    public void indexAndTextIndexCanCoexist() throws QueryException {
         String indexName = im.ensureIndexed(Arrays.<FieldSort>asList(new FieldSort("name"), new FieldSort("age")),
                                             "textIndex",
                                             IndexType.TEXT,
-                                            settings);
+                                            "porter");
         assertThat(indexName, is("textIndex"));
         indexName = im.ensureIndexed(Arrays.<FieldSort>asList(new FieldSort("name"), new FieldSort("age")), "jsonIndex");
         assertThat(indexName, is("jsonIndex"));
-        Map<String, Map<String, Object>> indexes = im.listIndexes();
-        assertThat(indexes.keySet(), containsInAnyOrder("textIndex", "jsonIndex"));
+        List<Index> indexes = im.listIndexes();
+        assertThat(indexes, containsInAnyOrder(getIndexNameMatcher("textIndex"),
+                getIndexNameMatcher("jsonIndex")));
     }
 
-    @Test
-    public void correctlyLimitsTextIndexesToOne() {
+    @Test(expected = QueryException.class)
+    public void correctlyLimitsTextIndexesToOne() throws QueryException {
         String indexName = im.ensureIndexed(Arrays.<FieldSort>asList(new FieldSort("name"), new FieldSort("age")), "basic", IndexType.TEXT);
         assertThat(indexName, is("basic"));
         indexName = im.ensureIndexed(Arrays.<FieldSort>asList(new FieldSort("name"), new FieldSort("age")), "anotherIndex", IndexType.TEXT);
-        assertThat(indexName, is(nullValue()));
     }
 
     @Test
-    public void createIndexUsingNonAsciiText() {
+    public void createIndexUsingNonAsciiText() throws QueryException {
         // can create indexes successfully
         String indexName = im.ensureIndexed(Arrays.<FieldSort>asList(new FieldSort("اسم"), new FieldSort("datatype"), new FieldSort("ages")),
                                             "basic");
         assertThat(indexName, is("basic"));
     }
 
     @Test
-    public void normalizeIndexFields() {
-        HashMap<String, String> nameField = new HashMap<String, String>();
-        nameField.put("name", "asc");
-        HashMap<String, String> petField = new HashMap<String, String>();
-        petField.put("pet", "desc");
-
-        // removes directions from the field specifiers
-        List<String> fields;
-        fields = IndexCreator.removeDirectionsFromFields(Arrays.<FieldSort>asList(new FieldSort("name", FieldSort.Direction.ASCENDING),
-                                                                               new FieldSort("pet", FieldSort.Direction.DESCENDING),
-                                                                               new FieldSort("age")));
-        assertThat(fields, containsInAnyOrder("name", "pet", "age"));
-    }
+    public void createIndexWhereFieldNameContainsDollarSign() throws QueryException {
```

Comment:
I think this test needs to be spilt into 3, and use the expected class to verify the index isn't created for the on that should fail.